assets/js/modules/idea-hub/datastore/draft-post-ideas.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * `modules/idea-hub` data store: draft-post-ideas.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const fetchGetDraftPostIdeasStore = createFetchStore( {
+	baseName: 'getDraftPostIdeas',
+	controlCallback: () => {
+		return API.get( 'modules', 'idea-hub', 'draft-post-ideas' );
+	},
+	reducerCallback: ( state, draftPostIdeas ) => {
+		return {
+			...state,
+			draftPostIdeas,
+		};
+	},
+} );
+
+const baseInitialState = {
+	draftPostIdeas: undefined,
+};
+
+const baseResolvers = {
+	*getDraftPostIdeas( options = {} ) {
+		const registry = yield Data.commonActions.getRegistry();
+		const draftPostIdeas = registry.select( STORE_NAME ).getDraftPostIdeas( options );
+
+		// If there are already draft ideas in state, don't make an API request.
+		if ( draftPostIdeas?.length ) {
+			return;
+		}
+
+		yield fetchGetDraftPostIdeasStore.actions.fetchGetDraftPostIdeas();
```

Comment:
We should strictly check whether the `draftPostIdeas` is undefined to make sure we don't send unnecessary requests when `draftPostIdeas` is an empty array (which is a possible value).

Reply:
Can you check again @eugene-manuilov ?